magento/magento-cloud-docker
# MCLOUD-5859: Fix for conflicting volumes - #43

### Description

https://jira.corp.magento.com/browse/MCLOUD-5859
Currently, the Magento Cloud Docker uses magento-sync volume and other volumes with no prefix for project. This causes collision and conflict when trying to switch projects.

To solve this, it should use a prefix for all volumes, this should probably be from .magento.app.yaml or .magento.docker.yaml and is by default set as follows:name: mymagento

This would then create volumes as mymagento-magento-sync and mymagento-magento-db. This allows projects to set this name and have it generate volumes that can persist.


### Fixed Issues (if relevant)

1. magento/magento-cloud-docker#43: Give us the ability to have multiple projects with different volumes (Prefixed the volume name with the name that is provided in .magento.app.yaml or .magent.docker.yaml)
2. Made Name as a required field. (So that the volumes can prefixed with the given name)

### Manual testing scenarios

1) Add .magento.app.yaml or .magento.docker.yaml (with name field included).
2) Run ./vendor/bin/ece-docker build:compose --mode="developer".
3) docker-compose.yaml will be generated.
4) Verify that volume names are prefixed with Name given in .magento.app.yaml or .magento.docker.yaml
5) Verify Multilple projects with different volumes are working.
     
- Mount a project using docker environment
- After that, try to create another one using the same architecture

7) Both projects should work with no error
8) if no Name is provided in .magento.app.yaml or .magento.docker.yaml an  exception will be thrown "Name could not be parsed."

### Contribution checklist

releasenotes
Fixed an issue that can cause volume conflicts when using multiple Docker environments.
